<h1>DSCI 100 Project Individual Planning</h1>

**Chanyoung Park 43150432 Section 009 Group 9**

In [1]:
library(tidyverse)
library(repr)
library(tidymodels)
library(cowplot)
options(repr.matrix.max.rows = 6)
source('cleanup.R')

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


In [3]:
#Reading Data Tables
players <- read_csv("data/players.csv")
sessions <- read_csv("data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [20]:
sessions <- sessions |>
    mutate(original_end_time = as.numeric(as.POSIXct(end_time, format = "%d/%m/%Y %H:%M")),
           original_start_time = as.numeric(as.POSIXct(start_time, format = "%d/%m/%Y %H:%M")))

In [21]:
#summary statistics of each data table
summary_players <- summary(players)
summary_players
summary_sessions <- summary(sessions)
summary_sessions

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+09  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+09  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+09  
                                                          Mean   :1.719e+09  
                                                          3rd Qu.:1.722e+09  
                                                          Max.   :1.727e+09  
                                                                             
 original_end_time  
 Min.   :1.712e+09  
 1st Qu.:1.716e+09  
 Median :1.719e+09  
 Mean   :1.719e+09  
 3rd Qu.:1.722e+09  
 Max.   :1.727e+09  
 NA's   :2          

**<h3>List of Variables in players.csv</h3>**
- **experience** : character type vector, Gaming experience level of each player
- **subscribe** : logical type vector, Whether the player subscribed to the Minecraft-related newsletter (TRUE/FALSE)
- **hashedEmail** : character type vector, An identifier for each player
- **played_hours** : double (numeric) type vector, Total hours the player has spent on the server
- **name** :  character type vector, A name of each player
- **gender** : character type vector, A gender of player
- **Age** : double (numeric) type vector, Age in years

**<h3>List of Variables in sessions.csv</h3>**
- **hashedEmail** : character type vector, An identifier for each player
- **start_time** : character type vector, The human-readable timepoint when a player started playing game
- **end_time** : character type vector, The human-readable timepoint when a player stopped playing game
- **original_start_time** : double (numeric) type vector, A timepoint when a player started playing game, which was recorded by the server
- **original_end_time** :  double (numeric) type vector, A timepoint when a player stopped playing game, which was recorded by the server

In [19]:
# Summary statistics for Experience and Gender in plyaer.csv
summary_experience <- players |>
  count(experience, name = "Count") |>
  arrange(desc(Count))
summary_experience

summary_gender <- players |>
  count(gender, name = "Count") |>
  arrange(desc(Count))
summary_gender

experience Count
1 Amateur    63   
2 Veteran    48   
3 Regular    36   
4 Beginner   35   
5 Pro        14

gender       Count
1 Male         124  
2 Female        37  
3 Non-binary    15  
⋮ ⋮            ⋮    
5 Two-Spirited 6    
6 Agender      2    
7 Other        1

**<h3>Summary Statistics in players.csv</h3>**

| Variable | Statistic | Value |
|----------|-----------|--------|
| experience | Mode | Amateur |
| subscribe | Mode | True |
| played_hours | Mean | 5.846 |
| played_hours | Median | 0.10 |
| played_hours | Max | 223.10 |
| gender | Mode | Male |
| Age | Mean | 21.14 |
| Age | Min | 9 |
| Age | Max | 58 |

**<h3>Summary Statistics in sessions.csv</h3>**
| Variable | Statistic | Value |
|----------|-----------|--------|
| original_start_time | Mean | 1.719e+09 |
| original_start_time | Min | 1.712e+09 |
| original_start_time | Median | 1.719e+09 |
| original_start_time | Max | 1.727e+09 |
| original_end_time | Mean | 1.719e+09 |
| original_end_time | Min | 1.712e+09 |
| original_end_time | Median | 1.719e+09 |
| original_end_time | Max | 1.727e+09 |

**<h3>Data Description</h3>**
For the player.csv data, there are 7 variables with a total number of 196 observations. The average age of the players is 21.14, with minimum age of 9 and maximum age of 58. The most common gender of player is male, and there are more players subscribed to the newsletter than those who didn’t. The most common experience level of players is Amateur. The mean value for total play time is 5.846 hours, but this has a significant difference with median total play time of 0.1 hours. This issue in the data can be explained by the maximum total play time of 223.1 hours, suggesting inflation in the mean total play time by few players with longer playtime, while most of the players quit around 0.1 hours. The experience variable from the player is self-reported, which results in a gap between reported skill level of the player and their actual gaming skill. Thus, the experience variable is not the most accurate measure of a player's skill. (173 words)

**<h3>Project Questions</h3>**
